src/index.js

```diff
@@ -53,24 +53,44 @@ const createRunner = ({ runtime = "worker_threads" } = {}) =>
      * @param {*} onResult
      * @param {*} onFailure
      */
-    runTests(tests, watcher, onStart, onResult, onFailure) {
+    async runTests(tests, watcher, onStart, onResult, onFailure) {
       const { updateSnapshot, testNamePattern, maxWorkers } = this._config;
 
       if (!this._runInBand && this._isProcessRunner) {
-        return pMap(
+        const pool = this._pool;
+        const workers = new WeakSet();
+
+        await pMap(
           tests,
-          test => {
+          async test => {
             onStart(test);
 
-            return this._pool
-              .run({ test, updateSnapshot, testNamePattern })
-              .then(
-                result => void onResult(test, result),
-                error => void onFailure(test, error),
-              );
+            try {
+              const result = await pool.run({
+                test,
+                updateSnapshot,
+                testNamePattern,
+              });
+              onResult(test, result);
+            } catch (error) {
+              onFailure(test, error);
+            }
+
+            for (const worker of this._pool.threads) {
+              if (!workers.has(worker)) {
+                workers.add(worker);
+                // Use `process.disconnect()` instead of `process.kill`, so we can collect coverage
+                // See https://github.com/nicolo-ribaudo/jest-light-runner/issues/90#issuecomment-2812473389
+                worker.process.kill = worker.process.disconnect;
+              }
+            }
```

Comment:
Can we move them before `await pool.destroy()` and remove `workers`?

Reply:
Current code seems not working on CI, but it did work on my machine.

Reply:
Are you running Windows? I suspect it may be system related.

Reply:
Yes, I'm on Windows.

Reply:
Still nothing on Windows https://github.com/nicolo-ribaudo/jest-light-runner/actions/runs/14520035657/job/40738481319?pr=92#step:7:1365

Reply:
Tested on Linux on gitpod.io, works too.

Reply:
> Can we move `worker.process.kill = worker.process.disconnect` before `await pool.destroy()` and remove `workers`?

Looks like move the end works, but I'm not sure if it will be called internally before we call `pool.destroy()`.

Reply:
As of now, no, it just calls `process.unref().`

Reply:
Look like the subprocess will die, I tested `isolateWorkers: true`

https://github.com/fisker/jest-light-runner/actions/runs/14521140796/job/40742114663?pr=92#step:7:1376

Reply:
But you are right, collecting workers won't help.

Can you help me take another look? I'll merge if the current code looks good to you.

Thanks a lot for the help!